crypto/ec/ec_curve.c

```diff
@@ -3246,3 +3248,158 @@ int EC_curve_nist2nid(const char *name)
     }
     return NID_undef;
 }
+
+static
+int ec_group_get_curve(const EC_GROUP *group, BIGNUM *p, BIGNUM *a,
+                       BIGNUM *b, BN_CTX *ctx)
+{
+    int field_nid;
+
+    field_nid = EC_METHOD_get_field_type(EC_GROUP_method_of(group));
+
+#ifndef OPENSSL_NO_EC2M
+    if (field_nid == NID_X9_62_characteristic_two_field) {
+        return EC_GROUP_get_curve_GF2m(group, p, a, b, ctx);
+    } else
+#endif /* !def(OPENSSL_NO_EC2M) */
+    if (field_nid == NID_X9_62_prime_field) {
+        return EC_GROUP_get_curve_GFp(group, p, a, b, ctx);
+    } else {
+        /* this should never happen */
+        return 0;
+    }
+}
+
+static
+int ec_point_get_affine_coordinates(const EC_GROUP *group,
+                                    const EC_POINT *point, BIGNUM *x,
+                                    BIGNUM *y, BN_CTX *ctx)
+{
+    int field_nid;
+
+    field_nid = EC_METHOD_get_field_type(EC_GROUP_method_of(group));
+
+#ifndef OPENSSL_NO_EC2M
+    if (field_nid == NID_X9_62_characteristic_two_field) {
+        return EC_POINT_get_affine_coordinates_GFp(group, point, x, y, ctx);
```

Comment:
I just realized I inverted the names in this function and I am calling the GF2m function instead of the GFp and vice-versa.

I am fixing this in the upcoming fixup commit that moves this code in `ec_lib.c`

Reply:
Fixed in 06dc1c4bc60b4f79b1a00fe277791a261b04437e

Reply:
Ok. Let me know when you're ready. I will need to push this now, since the repo is frozen.